Develop

amchikin opened the pull request:
Коммиты для обсуждения.

amchikin committed:
Тестовый коммит

amchikin committed:
Add codeconventions

amchikin committed:
Поработал с обработкой ошибок. @ControllerAdvice

amchikin committed:
Add MapStruct. Delete ModelMapper.

amchikin committed:
Add MapStruct. Delete ModelMapper.

amchikin committed:
Add @RequiredArgsConstructor from Lombok

amchikin committed:
Изменения для API showCards - добавил отображение имени и фамилии (пока в двух разных строках)

amchikin committed:
Изменения для API showCards - добавил отображение имени и фамилии (пока в двух разных строках)

amchikin committed:
Изменения для API showCards: добавлен MapperService. API cделано полностью по ТЗ. Плюс правки по Warnings

amchikin committed:
API добавления новой картой с привязкой к человеку, валидация.

bogdanbolshakov commented on src/main/java/ru/example/BankCard/advice/Advice.java:
```diff
@@ -0,0 +1,36 @@
+@ControllerAdvice
+public class Advice {
+
+    @ExceptionHandler  // метод который, ловит исключения и возвращает необходимый объект
```
комментарии пишутся не сбоку, а используя javadoc
привыкай писать нормально сразу
для описания методов используй /* + enter

amchikin replied:
Fix

bogdanbolshakov commented on src/main/java/ru/example/BankCard/advice/Advice.java:
```diff
@@ -0,0 +1,36 @@
+import ru.example.BankCard.exception.*;
+
+@ControllerAdvice
+public class Advice {
```
обычно принято называть такие классы ExceptionHandler

amchikin replied:
Fix

bogdanbolshakov commented on src/main/java/ru/example/BankCard/advice/Advice.java:
```diff
@@ -0,0 +1,36 @@
+
+    @ExceptionHandler  // метод который, ловит исключения и возвращает необходимый объект
+    private ResponseEntity<PeopleErrorResponse> handleException(PersonNotFoundException e) {
+        PeopleErrorResponse response = new PeopleErrorResponse(
```
существует аннотация @ builder и, в целом, было бы неплохо разобраться в паттерне билдер/построитель, и попробовать написать свой )
тут используй builder 
также старайся избегать создания объектов, особенно dto, через new

amchikin replied:
В паттерне разобрался: почитал теорию, подсмотрел пару реализаций, свою не писал. Как использовать понял и применил.

bogdanbolshakov commented on src/main/java/ru/example/BankCard/advice/Advice.java:
```diff
@@ -0,0 +1,36 @@
+        PeopleErrorResponse response = new PeopleErrorResponse(
+                "Person with current id does not exist in the database", System.currentTimeMillis()
+        );
+        return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
```
используй билдер + сразу возвращай объект, код должен быть кратким и лаконичным

amchikin replied:
Done

bogdanbolshakov commented on src/main/java/ru/example/BankCard/advice/Advice.java:
```diff
@@ -0,0 +1,36 @@
+    }
+    @ExceptionHandler
+    private ResponseEntity<PeopleErrorResponse> handleException(PersonNotCreateException e) {
+        PeopleErrorResponse response = new PeopleErrorResponse(  //
```
тут тоже билдер

amchikin replied:
Done

bogdanbolshakov commented on src/main/java/ru/example/BankCard/advice/Advice.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @ExceptionHandler
+    private ResponseEntity<AccountErrorResponse> handleException(AccountNotCreateException e) {
```
билдер
+ javadoc к методу оставь, а не так, как тут

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/AccountController.java:
```diff
@@ -0,0 +1,39 @@
+import ru.example.BankCard.service.AccountService;
+
+import java.util.List;
+@RequiredArgsConstructor
```
принято аннотацию restcontroller в контроллерах указывать в самом верху

amchikin replied:
То есть самую первую из аннотаций сразу после всех импортов? Сделал так:
......
import ru.example.BankCard.service.AccountService;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/accounts")
public class AccountController {
......

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/AccountController.java:
```diff
@@ -0,0 +1,39 @@
+
+    @GetMapping()
+    public List<AccountDto> getAccounts() {
+        return accountService.findAll();
```
findAll - это декларативный подход в JPA
тут у тебя метод из сервиса, дай ему нормальное бизнесовое имя, что-то типо getAccountList

amchikin replied:
Done

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/AccountController.java:
```diff
@@ -0,0 +1,39 @@
+    @PostMapping
+    public ResponseEntity<HttpStatus> create(@RequestBody @Valid AccountSaveDto accountSaveDto,
+                                             BindingResult bindingResult) {
+        AccountErrorResponse.CreateErrors(bindingResult);
```
что такое bindingResult? зачем он тут?

amchikin replied:
Объяснял в разговоре, решили что нужно убрать - убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/AccountController.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @PostMapping
+    public ResponseEntity<HttpStatus> create(@RequestBody @Valid AccountSaveDto accountSaveDto,
```
метод create, dto AccountSaveDto - ты уж определись, обычно в контроллерах create, save - опять же из jpa
плюс идеальное название дто - CreateAccountRqDto (создать аккаунт реквест дто - читается как книга и из названия сразу все понятно)

amchikin replied:
done

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/PeopleController.java:
```diff
@@ -0,0 +1,49 @@
+@RequestMapping("/people")
+public class PeopleController {
+
+    private final PeopleService peopleService; // Поле внедряется через конструктор авто средствами Lombok
```
коммент ненужный

amchikin replied:
Знаю, он был для меня больше. Убрал)

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/PeopleController.java:
```diff
@@ -0,0 +1,49 @@
+
+    @GetMapping()
+    public List<PersonDto> getPeople() {
+        return peopleService.findAll();
```
нейминг сервисного метода очень плохой

amchikin replied:
Поправил

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/PeopleController.java:
```diff
@@ -0,0 +1,49 @@
+        return peopleService.findAll();
+    }
+
+    // API 1 по ТЗ: Человек по id. Если нет нашёл - читаемое исключение.
```
преобразуй в javadoc

amchikin replied:
Есть

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/PeopleController.java:
```diff
@@ -0,0 +1,49 @@
+    // API 1 по ТЗ: Человек по id. Если нет нашёл - читаемое исключение.
+    @GetMapping("/{id}")
+    public PersonDto getPerson(@PathVariable("id") int id) {
+        return peopleService.findOne(id);
```
нейминг - peopleService.getPersonById(id)
ну гораздо же читаемое, разве нет?

amchikin replied:
done

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/PeopleController.java:
```diff
@@ -0,0 +1,49 @@
+
+    // API 1 по ТЗ: Человек по id. Если нет нашёл - читаемое исключение.
+    @GetMapping("/{id}")
+    public PersonDto getPerson(@PathVariable("id") int id) {
```
ну и конечно id лучше никогда не делать int, обсуждали

amchikin replied:
Виновен - просмотрел. Отголоски разработки на С++. Исправил.

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/PeopleController.java:
```diff
@@ -0,0 +1,49 @@
+    // API 2 по ТЗ. Выводит инфо о всех картах человека по id (сортирует по возрастанию баланса).
+    @GetMapping("/{id}/cards")
+    public ShowCardsResponseDto showCards(@PathVariable("id") int id) {
+        return peopleService.getCardsByPersonId(id);
```
вот тут за нейминг похвала и уважение )

amchikin replied:
Благодарю!

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/PeopleController.java:
```diff
@@ -0,0 +1,49 @@
+    @PostMapping
+    public ResponseEntity<HttpStatus> create(@RequestBody @Valid PersonDto personDTO,
+                                             BindingResult bindingResult) {
+        PeopleErrorResponse.CreateErrors(bindingResult);
```
тот же про binding result - что это и зачем?

amchikin replied:
Убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/dto/AbstractDto.java:
```diff
@@ -0,0 +1,6 @@
+
+import java.io.Serializable;
+
+public abstract class AbstractDto implements Serializable {
```
redundant, или по русски - нахер он нужен?)

amchikin replied:
Сериализация тут не нужна - убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/dto/AccountSaveDto.java:
```diff
@@ -0,0 +1,28 @@
+import java.math.BigInteger;
+
+
+@Getter
```
когда ты используешь getter и setter, то есть аннотация, объединяющая их, и не только - @data

amchikin replied:
done

bogdanbolshakov commented on src/main/java/ru/example/BankCard/dto/AccountSaveDto.java:
```diff
@@ -0,0 +1,28 @@
+
+    private Boolean isSalary;
+
+    private Integer person_id;
```
так поля называть нельзя, читай конвенции которые скидывал тебе

amchikin replied:
"Ignore the rules" )
Переименовал.

bogdanbolshakov commented on src/main/java/ru/example/BankCard/dto/ShowCardsResponseDto.java:
```diff
@@ -0,0 +1,13 @@
+
+import java.util.List;
+
+@Getter
```
аннотация data + зачем ты это наследуешь?)

amchikin replied:
fix

bogdanbolshakov commented on src/main/java/ru/example/BankCard/entity/Account.java:
```diff
@@ -1,13 +1,15 @@
+@Getter
+@Setter
 @Entity
 @Table(name="Account")
```
либо все с большой, либо все с маленькой
я про название таблицы

amchikin replied:
Есть

bogdanbolshakov commented on src/main/java/ru/example/BankCard/entity/Person.java:
```diff
@@ -1,16 +1,18 @@
+@Getter
+@Setter
 @Entity
 @Table(name="Person")
```
аналогично, либо большими либо маленькими

amchikin replied:
Есть

bogdanbolshakov commented on src/main/java/ru/example/BankCard/entity/Person.java:
```diff
@@ -31,7 +33,7 @@ public class Person extends AbstractEntity {
 
     @Column(name="birthday")
-    @Temporal(TemporalType.DATE)
+    @Temporal(TemporalType.DATE) // TODO можно ли без неё
```
можно без нее

amchikin replied:
Убрал, проверил, да можно без неё

bogdanbolshakov commented on src/main/java/ru/example/BankCard/exception/AccountErrorResponse.java:
```diff
@@ -0,0 +1,30 @@
+    private String message;
+    private long timestamp;
+
+    public static void CreateErrors(BindingResult bindingResult) {
```
тебя exceptionHandler чем не устраивает?

amchikin replied:
Убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/exception/PeopleErrorResponse.java:
```diff
@@ -0,0 +1,30 @@
+    private long timestamp;
+
+    public static void CreateErrors(BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
```
а это уже дублирование кода, да еще и ненужного )

amchikin replied:
Убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapper/AccountMapper.java:
```diff
@@ -0,0 +1,12 @@
+@Mapper(componentModel = "spring")
+
+public interface AccountMapper {
+    Account toModel(AccountDto source);
```
обычно методы называют просто map

amchikin replied:
Таки да

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapper/AccountSaveMapperInjectService.java:
```diff
@@ -0,0 +1,23 @@
+@Mapper(componentModel = "spring")
+public abstract class AccountSaveMapperInjectService {
+    @Autowired  // TODO Почему подчёркивает и как сделать лучше
+    protected AccountSaveMapperService accountSaveMapperService;
```
тут этого быть не должно
это маппинг слой, никакой логики здесь быть не может
подумай над другим решением

amchikin replied:
Мапперы поправлены

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapper/AccountSaveMapperInjectService.java:
```diff
@@ -0,0 +1,23 @@
+import ru.example.BankCard.mapperService.AccountSaveMapperService;
+
+@Mapper(componentModel = "spring")
+public abstract class AccountSaveMapperInjectService {
```
это маппер, а не сервис ))

amchikin replied:
Исправлено)

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapper/ShowCardsResponseMapperInjectService.java:
```diff
@@ -0,0 +1,22 @@
+@Mapper(componentModel = "spring")
+public abstract class ShowCardsResponseMapperInjectService {
+    @Autowired  // TODO Почему подчёркивает и как сделать лучше
+    protected ShowCardsResponseService showCardsResponseService;
```
опять же, это маппер, тут никаких сервисов быть не должно
у тебя вся логика есть в сервисе, оттуда просто то что нужно в маппер отправляй

amchikin replied:
Убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapperService/AccountSaveMapperService.java:
```diff
@@ -0,0 +1,19 @@
+        this.peopleRepository = peopleRepository;
+    }
+
+    public Person ownerById(Integer person_id) {
```
почему это не в бизнес сервисе?
название у класса сбивает с толку - это маппер или сервис?

amchikin replied:
Поправлено

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapperService/AccountSaveMapperService.java:
```diff
@@ -0,0 +1,19 @@
+    }
+
+    public Person ownerById(Integer person_id) {
+        return peopleRepository.findById(person_id).get();
```
.get() - за такое бьют по рукам
научись обрабатывать null pointer exception, делай проверку на null, и либо возвращай значение либо кидай исключение

amchikin replied:
Убрал, не бейте)

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapperService/ShowCardsResponseService.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.stereotype.Service;
+
+@Service
+public class ShowCardsResponseService {
```
ну это уже перебор с сервисами)
у тебя должен быть один сервис, который работает с картами
там ты делаешь запросы в бд, и данные как-то обрабатываешь и маппишь
тут ты ж целый отдельный сервис для сложения строк придумал )
как сказала бы Наталья Андреевна Денисова : "мудрец"

amchikin replied:
Д, убрал, убрал)

bogdanbolshakov commented on src/main/java/ru/example/BankCard/mapperService/ShowCardsResponseService.java:
```diff
@@ -0,0 +1,11 @@
+@Service
+public class ShowCardsResponseService {
+
+    public String enrichName(String name, String surname) {
```
enrich - хуич
это конкатенация называется )

amchikin replied:
Ржал 5 минут) Убрал)

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/AccountService.java:
```diff
@@ -0,0 +1,14 @@
+
+import java.util.List;
+
+public interface AccountService {
```
у тебя у одного метода есть модификатор доступа, у другого нет
как правильно то должно быть?)

amchikin replied:
убрал модификатор доступа у методов.

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/AccountServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.List;
+@RequiredArgsConstructor
+@Service
```
сервис должен быть наверху

amchikin replied:
Сделано

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/AccountServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+    @Override
+    public List<AccountDto> findAll() {  // TODO Подумать над лучшей реализацией
+        List<AccountDto> listAccountDTO = new ArrayList<>();
+        List<Account> listAccount = accountsRepository.findAll();
```
используй стрим тут, подходит идеально

amchikin replied:
Да, идеально! одна строка)
    @Override
    public List<AccountDto> getAccountList() {
        return accountsRepository.findAll().stream().map(accountMapper::map).collect(Collectors.toList());
    }

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/AccountServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+    @Override
+    public void save(AccountSaveDto accountSaveDto) {
+        //return accountSaveMapper.toDto(accountsRepository.save(accountSaveMapper.toModel(accountSaveDto)));
+        accountsRepository.save(accountSaveMapperInjectService.ToModel(accountSaveDto));
```
намудрил тут опять
сделай просто сейв ентити да и все

amchikin replied:
Не совсем тебя тут понял, но переделал.

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/PeopleService.java:
```diff
@@ -0,0 +1,20 @@
+
+    List<PersonDto> findAll();
+
+    PersonDto findOne (int id);
```
следи за форматированием
пробела быть тут не должно
на маке есть hotkey cmd + opt + L для выравнивания кода
погугли какой хоткей на винде для этого

amchikin replied:
Сделал. Ctrl+Alt+L

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/PeopleServiceImpl.java:
```diff
@@ -0,0 +1,56 @@
+public class PeopleServiceImpl implements PeopleService {
+    private final PeopleRepository peopleRepository;
+    private final PersonMapper personMapper;
+    private final ShowCardsResponseMapperInjectService showCardsResponseMapperInjectService;
```
это ерунда тебе не нужна, избавься от нее

amchikin replied:
Убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/PeopleServiceImpl.java:
```diff
@@ -0,0 +1,56 @@
+    @Override
+    public List<PersonDto> findAll() {  // TODO Подумать над лучшей реализацией. Через Stream
+        List<PersonDto> listPersonDTO = new ArrayList<>();
+        List<Person> listPerson = peopleRepository.findAll();
```
через стрим

amchikin replied:
+

bogdanbolshakov commented on src/main/sql/db.sql:
```diff
@@ -1,14 +1,14 @@
+                        BIRTHDAY DATE
 );
 
 CREATE TABLE Account (
```
а почему account решил не делать в uppercase?

amchikin replied:
Поправил

bogdanbolshakov commented on src/main/java/ru/example/BankCard/service/PeopleServiceImpl.java:
```diff
@@ -0,0 +1,56 @@
+
+    @Override
+    public ShowCardsResponseDto getCardsByPersonId(int id) { // TODO
+        Optional<Person> person = peopleRepository.findById(id);
```
обработай optional в репо слое
если вернется объект null - кидай exception
дальше работай с картами человека через стрим
в начале лучше сделать проверку:
if(person.getAccounts() != null) {
    return person.getAccounts().stream()....дальше сам
}

amchikin replied:
Сделал по-другому, прошу дискуссию)

bogdanbolshakov reviewed (commented):
общее впечатление 
из плюсов - разобрался с обработкой ошибок, появился mapstruct
из минусов - наплодил кучу сервисов
еще раз, общая логика следующая
запрос пришел в контроллер -> вызываешь бизнес сервис -> внутри сервиса: 
а) через репо слой взаимодействуешь с бд
б) маппинг в энтити и обратно
в) бизнес логика из ТЗ

amchikin committed:
API 3: изменение баланса зарплатной карты по id человека(пока без проверки на уход в минусовой баланс). Логика нахождения зарплатной карты отвратительна - переделать.

amchikin committed:
Ничего не добавил нового, "подмёл" проект - почистил

amchikin committed:
API 3: добавил проверку на уход в минусовый баланс - выбрасывает исключение: "The balance cannot be negative"

amchikin committed:
Никакого функционала. Фиксирую TODO`s по итогу митинга.

amchikin committed:
Поработал с мапперами

amchikin committed:
Поработал с мапперами ещё

amchikin committed:
Рефакторинг: имена, комментарии

amchikin committed:
Добавил @Builder

amchikin committed:
Добавил красивые выводы в Постман на пост-запросы

amchikin committed:
Advice -> ExceptionError (Rename). Code format.

amchikin committed:
Исправление некоторых замечаний

amchikin committed:
Исправление некоторых замечаний 2

amchikin committed:
Исправление некоторых замечаний 3

amchikin committed:
Убрал не нужный @Temporal(TemporalType.DATE)

amchikin committed:
Поработал с исключениями: убрал BindingResult, убрал лишние классы и прочее

amchikin committed:
Вернул @Temporal в class Person - без него формат даты в Postman неверен - разобраться

amchikin committed:
Разобрался со Стримами, нормально переписал сервисные функции: больше синтаксического сахара, меньше соли)

bogdanbolshakov commented on src/main/java/ru/example/BankCard/controller/AccountController.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @PostMapping()
+    public String createAccount(@RequestBody @Valid AccountSaveDto accountSaveDto) {
```
ты возвращаешь стринги в контроллере, это не json формат
надо dto тоже возвращать

amchikin replied:
Могу сделать через ResponseEntity(на картинке показал. Закомментировано то, как сделано сейчас).  Но тогда сообщение не вставить. А как вообще в интеллигентном обществе делают?)
![Снимок](https://github.com/amchikin/BankCard/assets/152810620/60a19494-2cd7-468b-8a3f-a67581e45a3c)

amchikin replied:
Сделано

bogdanbolshakov commented on src/main/java/ru/example/BankCard/entity/Account.java:
```diff
@@ -7,31 +7,27 @@
-
     @ManyToOne
-    @JoinColumn(name = "person_id", referencedColumnName = "id")
+    @JoinColumn(name = "personid", referencedColumnName = "id")
```
personId, читай конвенции

amchikin replied:
Сделал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/entity/Person.java:
```diff
@@ -12,31 +12,25 @@
-    @Column(name="birthday")
-    @Temporal(TemporalType.DATE)
+    @Column(name = "birthday")
+    @Temporal(TemporalType.DATE) // TODO Без неё в Postman формат даты неверен
```
а что в постмане отправляешь?)

amchikin replied:
В Постман при создании сущности делаю запрос так:
{
    "surname": "Hubba",
    "name": "Bubba",
    "birthday": "2010-01-01"
}

Я перешёл на тип даты LocalDate - проблема ушла, **@Temporal убрал** (про @Temporal почитал, эта аннотация тут, действительно, не к месту)

bogdanbolshakov commented on src/main/java/ru/example/BankCard/exception/ControllerExceptionHandler.java:
```diff
@@ -0,0 +1,44 @@
+public class ControllerExceptionHandler {
+    @ExceptionHandler(NotFoundException.class)
+    private ResponseEntity<ErrorMessage> resourceNotFoundException(NotFoundException ex) {
+        final ErrorMessage message = new ErrorMessage(
```
зачем final здесь?

amchikin replied:
Не нужно, убрал

bogdanbolshakov commented on src/main/java/ru/example/BankCard/exception/ErrorMessage.java:
```diff
@@ -0,0 +1,18 @@
+    private String message;
+
+    public ErrorMessage(int statusCode, Date timestamp, String message) {
+        this.statusCode = statusCode;
```
используй allargsconstructor

amchikin replied:
Полностью согласен

amchikin committed:
1. Unit-tests - проба пера, начало
2. Data -> LocalDate
3. Enter between Entity fields

amchikin committed:
1. Unit-tests(Разобрался с применением Маппера)
2. Поправил имена в таблицы согласно конвенциям SQL

amchikin committed:
1. Исправления в рамках Pull request
2. Подружил @Builder и @Data

amchikin committed:
4 теста для сервисного слоя

amchikin committed:
5 тестов для 5 функций из сервисного слоя для Person. (Есть проблема для теста testGetCardsByPersonId)

amchikin committed:
1. Перешёл на application.yml
2. Liquibase: подключил, добавил добавление двух таблиц
3. Пятый тест, который не проходит пока закоментил

amchikin committed:
Liquibase: разобрался с добавлением данных

amchikin committed:
Развернул Postgres в Docker и подключил к Spring-Boot приложению

amchikin committed:
Откат LiquiBase, Docker

amchikin committed:
Правки для тестов

amchikin committed:
Поработал с ResponseEntity для AccountController

amchikin committed:
Add Response and Request fo PeopleSaveDto

amchikin committed:
Docker. Разобрался как сделать контейнеры для БД(postgres) и PgAdmin

amchikin committed:
Add Liquibase (рабочая версия)

amchikin committed:
Fix баг с ответом для функции ShowCardsDto getCardsByPersonId(Integer id)

amchikin committed:
Почистил проект, поправил один тест

amchikin committed:
Вариант решения проблемы n+1 для функции getCardsByPersonId() через EntityManager

amchikin committed:
Решения проблемы n+1 для функции getCardsByPersonId() через @EntityGraph

amchikin committed:
Решения проблемы n+1 для функции getAccountList() через @EntityGraph

amchikin committed:
Добавил 3 новые API:
1) Покажи карту по id
2) Измени человека по id
3) Измени карту по id
